examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart

```diff
@@ -0,0 +1,565 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+class BottomAppBarDemo extends StatefulWidget {
+  static const String routeName = '/material/bottom_app_bar';
+
+  @override
+  State createState() => new _BottomAppBarDemoState();
+}
+
+class _BottomAppBarDemoState extends State<BottomAppBarDemo> {
+  // The key given to the Scaffold so that _showSnackbar can find it.
+  static final GlobalKey<ScaffoldState> _scaffoldKey = new GlobalKey<ScaffoldState>();
+
+  // The index of the currently-selected _FabLocationConfiguration.
+  int fabLocationIndex = 1;
+
+  static const List<_FabLocationConfiguration> _fabLocationConfigurations = const <_FabLocationConfiguration>[
+    const _FabLocationConfiguration('End, undocked above the bottom app bar', _BabMode.END_FAB, FloatingActionButtonLocation.endFloat),
+    const _FabLocationConfiguration('End, docked to the bottom app bar', _BabMode.END_FAB, FloatingActionButtonLocation.endDocked),
+    const _FabLocationConfiguration('Center, docked to the bottom app bar', _BabMode.CENTER_FAB, FloatingActionButtonLocation.centerDocked),
+    const _FabLocationConfiguration('Center, undocked above the bottom app bar', _BabMode.CENTER_FAB, FloatingActionButtonLocation.centerFloat),
+    // This configuration uses a custom FloatingActionButtonLocation.
+    const _FabLocationConfiguration('Start, docked to the top app bar', _BabMode.CENTER_FAB, const _StartTopFloatingActionButtonLocation()),
+  ];
+  
+  // The index of the currently-selected _FabShapeConfiguration.
+  int fabShapeIndex = 1;
+
+  final List<_FabShapeConfiguration> _fabShapeConfigurations =  <_FabShapeConfiguration>[
+      const _FabShapeConfiguration('None', null),
+      new _FabShapeConfiguration('Circular', 
+        new FloatingActionButton(
+          onPressed: () => _showSnackbar(),
+          child: const Icon(Icons.add),
+          backgroundColor: Colors.orange,
+        ),
+      ),
+      new _FabShapeConfiguration('Diamond',
+        new _DiamondFab(
+          onPressed: () => _showSnackbar(),
```

Comment:
This can just be `onPressed: _showSnackBar`

Reply:
Tearoff applied